modules/imgproc/src/color_yuv.cpp

```diff
@@ -1108,16 +1270,77 @@ struct YUV420p2RGB8Invoker : ParallelLoopBody
             uchar* row1 = dst_data + dst_step * j;
             uchar* row2 = dst_data + dst_step * (j + 1);
             const uchar* y2 = y1 + stride;
+            int i = 0;
+
+#if CV_SIMD
+            const int vsize = v_uint8::nlanes;
+            v_uint8 a = vx_setall_u8(uchar(0xff));
+            for( ; i <= width/2 - vsize;
+                 i += vsize, row1 += vsize*dcn*2, row2 += vsize*dcn*2)
+            {
+                v_uint8 u, v;
+                u = vx_load(u1 + i);
+                v = vx_load(v1 + i);
```

Comment:
What are the alignment requirements of these(any and all vector) instructions and the guarantees the API/users give? _this applies to all the other places in this and similar PRs_

There are platforms(e.g. ARM) that do not support unaligned instructions(even regular scalar non-vector ones), others have big performance hit(e.g. 2x on x86/64)

Were there any tests on non-x86 platforms?

--
Specifically, a lot of people(including us) are having this great signal 7 (SIGBUS), code 1 (BUS_ADRALN) crashes all over the place on android(and less often on ios), some devices/os versions more prone(100%) than the others.
Ours particular reproducible crash is reading VideoCapture->yuv420 conversion.

--
@alalek It raises an eyebrow how fast this PR was merged(even with a single review).
Usually vectorised code is a bugs' nest and it raises a question on the quality of this and similar PRs
I'm new to the opencv processes but i wonder if there are any people at intel/opencv team that would like to review this/similar PRs more thoroughly. What is the process to finding these people? (sorry have not had a time to read any contribution guidelines yet)

--
The usual process to vectorising the code is having:
- unvectorised prologue for unaligned part
- vectorised loop for aligned & full blocks
- unvectorised epilogue for the rest of incomplete block

this code(and probably other places and similar PRs) seem to miss the first part completely.

--
It is better to have a working code than fast but incorrect(and even crashing) one. 
Is there a possibility to review all this vectorised "mess" for alignment and fix it quickly, or is it better to revert completely until better-thought-out version arrives?

Reply:
```
if(range.start % 2 == 1)
        {
            u1 += uvsteps[(usIdx++) & 1];
            v1 += uvsteps[(vsIdx++) & 1];
        }
```
Was this a (historical) attempt at fixing alignment of (scalar) accesses back then..?

Reply:
There are two [intrinsic functions](https://docs.opencv.org/master/df/d91/group__core__hal__intrin.html) in OpenCV:
- `vx_load` (wrapper over `v_load()` (SIMD128) / `v256_load()` / `v512_load()`)
- `vx_load_aligned`

Both should be aligned on the size of base type (uchar is that case, so no extra limitation): [docs](https://docs.opencv.org/master/df/d91/group__core__hal__intrin.html#ga1edd512aeff776eb2e7e742983686d40)

---

Feel free to prepare reproducer for this problem and open new issue (in form of PR with test is preferable - we have ARM builders in CI (however there can be some extra settings near alignment requirements)).

Reply:
Big thanks for the detailed information as always!
Sorry for not reading more into intrinsics, i assumed they were a simple generic unaligned/untyped ones.

One possible reason might be the direct use of the Device(gpu/codec) memory which may have some quirks for wide accesses...
I'll try gathering more information on this.